src/Data/Monoid/Action.hs

```diff
@@ -81,3 +85,45 @@ instance Action m s => Action (Maybe m) s where
 instance Action (Endo a) a where
   act = appEndo
 
+instance Num a => Action Integer (Sum a) where
+  n `act` a = fromInteger n <> a
+
+instance Num a => Action Integer (Product a) where
+  n `act` a = fromInteger n <> a
+
+instance Fractional a => Action Rational (Sum a) where
+  n `act` a = Sum (fromRational n) <> a
+
+instance Fractional a => Action Rational (Product a) where
+  n `act` a = Product (fromRational n) <> a
+
+-- | An action of a group is "free transitive", "regular", or a "torsor"
+--   iff it is invertible.
+--
+--   Given a value `s2` that has been acted upon by `m`,
+--   and the original value `s1`, it is possible to recover the acting `m`.
+--   This is encoded in the laws:
```

Comment:
Thanks, I think this is now correct, but it is still a bit confusing because at first you say "given a value `s2` that has been acted upon by `m`, and the original value `s1`", but then later you use the variable names `s1` and `s2` to mean the opposite things, e.g. in ``(s1 `difference` s2) `act` s2 = s1``, it is `s2` which is the original value and `s1` the result of acting on `s2` by `m`.  Personally I think `s1` and `s2` should be swapped in the below laws and in the Haddock comment for `difference`.  But it would also be consistent to swap them in this initial explanatory comment.

Also, "a value `s2` that has been acted upon" is a bit ambiguous, does it mean that `s2` is the result of acting upon something by `m` (yes) or that we have computed ``m `act` s2`` (no, but "a value that has been acted upon" kind of sounds like that's what it means).  I suggest making it unambiguous by saying something like "given an original value `s1` and a value `s2` which is the result of acting upon `s1` by some `m` (i.e.  ``s2 = m `act` s1``), it is possible..."

Reply:
That's good feedback! I like good docs as well. I removed the numbers and replaced them by explanatory suffixes, maybe this is easier to understand?

Reply:
Yes, that's much better, thanks!